### PR TITLE
e2e: new ExitCleanly matcher

### PR DESCRIPTION
**Pull request by edsantiago**
e2e test logs are full of warning messages that we never see. Let's start looking for them.

This PR introduces a new gomega matcher, `ExitCleanly()`, with the purpose of replacing `Should(Exit(0))`. This matcher checks exit code _and_ that no unexpected messages are emitted on stderr. Checking stdout is still up to the test writer.

As proof of concept, use the new matcher in `play_build_test.go`. This resulted in tests failing, because all those `kube play` commands have been spitting out warnings for lo all these years:

     time=.... level=warning msg="HEALTHCHECK is not supported for OCI image format and will be ignored. Must use `docker` format"

To remove that warning, use our standard system-test `libpod/testimage:YYYYMMDD` image instead of `alpine_nginx`. Image is now fetched and cached as part of standard test prep; I expect to use it much more in the future.

FWIW I tried updating `play_kube_test.go` but discovered and filed #19801, a bug that makes that impossible.

```release-note
None
```

### CODE REVIEW

File: test/e2e/config_arm64.go [hunk 1]
```diff
@@ -5,11 +5,11 @@ var (
...
 	NGINX_IMAGE              = "quay.io/lsm5/alpine_nginx-aarch64:latest"                                                                                                                //nolint:revive,stylecheck
 	BB_GLIBC                 = "docker.io/library/busybox:glibc"                                                                                                                         //nolint:revive,stylecheck
 	REGISTRY_IMAGE           = "quay.io/libpod/registry:2.8.2"                                                                                                                           //nolint:revive,stylecheck
-	LABELS_IMAGE             = "quay.io/libpod/alpine_labels:latest"                                                                                                                     //nolint:revive,stylecheck
+	CITEST_IMAGE             = "quay.io/libpod/testimage:20221018"                                                                                                                       //nolint:revive,stylecheck
```
Comment (TomSweeneyRedHat) [inline test/e2e/config_arm64.go:12]: Two questions for further consideration.  They don't need to be addressed for this PR.

1) As this variable is referenced in at least two places, is there a place it could be added to and then pulled into here?

2) Just looking at the testimage date.  It's almost a year old, is it time to bump it?

**Comment by TomSweeneyRedHat** [comment]
LGTM
assuming happy tests

File: test/e2e/config_arm64.go (hunk 1 above)
Comment (edsantiago) [reply test/e2e/config_arm64.go:12]: Could you rephrase question 1? I can't find a way to understand it. I'm sorry.

Bumping: absent any pressing need, bumping doesn't really gain us anything...

**Comment by edsantiago** [comment]
Unsurprisingly, it got complicated. [f38 root containerized[(https://api.cirrus-ci.com/v1/artifact/task/4969105690722304/html/int-podman-fedora-38-root-container-boltdb.log.html) is showing a ton of:
```
           Unexpected warnings seen on stderr: "failed to connect to syslog: Initialization"
```
Looking deeper I see two things:
1. This is happening *all over*, not just the tests with the new `ExitCleanly()`; and
2. In one case, it's a flake. The test passes on second rerun.

This is not something I'll be able to fix.

File: test/e2e/play_build_test.go [hunk 2]
```diff
@@ -84,17 +83,17 @@ LABEL marge=mom
 		Expect(os.Chdir(yamlDir)).To(Succeed())
 		defer func() { (Expect(os.Chdir(cwd)).To(BeNil())) }()
 
-		session := podmanTest.Podman([]string{"play", "kube", "top.yaml"})
+		session := podmanTest.Podman([]string{"kube", "play", "top.yaml"})
```
Comment (vrothberg) [inline test/e2e/play_build_test.go:86]: I didn't check but please make sure that there are still some caller of `play kube` to make sure the alias continues to work.

**Review by vrothberg — COMMENTED** [review]
Nice work!
LGTM

**Review by flouthoc — APPROVED** [review]
LGTM other than comments above.

File: test/e2e/play_build_test.go (hunk 2 above)
Comment (edsantiago) [reply test/e2e/play_build_test.go:86]: Only a hundred and sixty-three instances.

**Comment by edsantiago** [comment]
Filed #19809 for the syslog bug. This is blocked until that gets resolved.

File: test/utils/matchers.go [hunk 3]
```diff
@@ -168,6 +168,57 @@ func (matcher *ExitMatcher) MatchMayChangeInTheFuture(actual interface{}) bool {
...
+		matcher.msg = fmt.Sprintf("Command failed with exit status %d", exitcode)
+		if stderr != "" {
+			matcher.msg += ". See above for error message."
+		}
+		return false, nil
+	}
+
+	// FIXME: #19809, "failed to connect to syslog" warnings on f38
+	// FIXME: so, until that is fixed, don't check stderr if containerized
+	if !Containerized() {
```
Comment (edsantiago) [inline test/utils/matchers.go:203]: Given:
1. I can't figure out where that warning is coming from; and
2. enabling _some_ testing is better than what we have now

...I've decided to kludge in a manual exception to skip warn-testing when running in container. 

@containers/podman-maintainers please take another look

**Review by vrothberg — COMMENTED** [review]
LGTM, great work!
